daemons/attrd/pacemaker-attrd.h

```diff
@@ -120,6 +120,7 @@ typedef struct attribute_s {
     char *uuid; /* TODO: Remove if at all possible */
     char *id;
     char *set;
+    char *block;
```

Comment:
I've always found the "set" terminology obscure, and now that we're adding block it looks even more so. I'm thinking something like block_type+block_id, set_type+set_id, or parent_type+parent_id.

Reply:
I don't think we can change the names of existing attributes in attrd messages, right?  Even though it's all internal, it'll cause problems for mixed version clusters.

With that in mind, we already have a->set = PCMK__XA_ATTR_SET = "attr_set".  I like keeping the structure name and the attribute name at least somewhat related so you don't have to think extra hard.  So maybe rename a->set to a->set_id, rename a->block to a->set_type, and renamed PCMK__XA_ATTR_BLOCK/"attr_block" to PCMK__XA_ATTR_SET_TYPE/"attr_set_type"?

Reply:
> I don't think we can change the names of existing attributes in attrd messages, right? Even though it's all internal, it'll cause problems for mixed version clusters.

Correct. It's never user-visible so we don't have to care about changing them. It's just the struct members so we get less confused :)

> 
> With that in mind, we already have a->set = PCMK__XA_ATTR_SET = "attr_set". I like keeping the structure name and the attribute name at least somewhat related so you don't have to think extra hard. So maybe rename a->set to a->set_id, rename a->block to a->set_type, and renamed PCMK__XA_ATTR_BLOCK/"attr_block" to PCMK__XA_ATTR_SET_TYPE/"attr_set_type"?

Sounds good